### PR TITLE
Fix GCSV2UnderFileSystem initialization

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

Fix GCSV2UnderFileSystem initialization

### Why are the changes needed?

not able to create ufs instance if not set access key, which should not be a requirement for gcs v2

### Does this PR introduce any user facing changes?

na

### CODE REVIEW

File: dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java
Comment (lucyge2022) [inline dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java:52]: why checkgcscredentials doesnt apply to this anymore?

Comment (jja725) [reply dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java:52]: GCS v2 only need to provide json file, only v1 is using access key. #17985 make the wrong check so I just revert it

Comment (lucyge2022) [reply dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java:52]: then why it was merged in the first place? sorry I don't have the context and I don't want to make the call of LGTM, if some bugs are introduced as part of other PR, can u plz let the related group know?

**Review by JiamingMai — APPROVED** [review]
LGTM

File: dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java
Comment (JiamingMai) [reply dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java:52]: I think this change is OK for GCS v2.

**Comment by jja725** [comment]
Tested locally and it works

**Comment by jja725** [comment]
alluxio-bot, merge this please

**Comment by jja725** [comment]
@zhezhidashi fixed here

**Pull request merged**
